### ISSUE TITLE
FracRidgeRegressor .fit() fails when data has only one column (ValueError: shape-mismatch for sum)

### ISSUE THREAD

**Issue by oliver-contier** [nrdg/fracridge#24]
I'm running into trouble when trying to use the `.fit()` method of FracRidgeRegressor where x has only one column.

```
fracs_ = np.arange(0., 1., .05)
fr = FracRidgeRegressor(fracs=fracs_, fit_intercept=True, normalize=True)

# x_train is array of shape (31240, 1)
# y_train is array of shape (31240, 211339)
fr.fit(x_train, y_train)

# *** ValueError: shape-mismatch for sum
```

I've tried removing redundant dimension with `x_train.squeeze()`, which only produced
```
*** ValueError: Expected 2D array, got 1D array instead:
array=[ 0.          0.          0.         ... -0.17669481 -0.05394767
 -0.01017657].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.
```

Interestingly, I run into the same problem when y has only one target. Using `fracridge.fracridge` instead of `fracridge.FracRidgeRegressor` seems to work though. Likewise, when x or y has more than one column, the `.fit` method works fine as well. Might this be an issue with the implementation, or am I misunderstanding something?

I'm currently trying this version pulled from github
```
import fracridge
fracridge.__version__
'1.3.2.dev364623090'
```

Cheers!

**Comment by arokem** [comment]
Hi @oliver-contier : thanks for reporting. I will investigate, but it might take a little bit of time, because I have some deadlines coming up in the next couple of weeks.

Question, because I am not sure that I understand the use-case: Why would you use ridge regression with only one regressor? Are you sure that X and Y are in the right order here? I'm only asking because the shapes look the opposite of what I'd usually expect. 

This should work, or at least provide a useful error.

**Comment by oliver-contier** [comment]
Hi @arokem,

No hurry, I switched to `fracridge.fracridge` for now and it seems to work fine. I was using the `FracRidgeRegressor` object for convenience; I also used its `.predict()` and `.score()` methods.

From the documentation and a glance at the code, I gather that the shape of x should be (nsamples, nregressors) and shape of y should be (nsamples, ntargets). And the `.fit()` method should take `.fit(x, y)` in that order. So I think the shapes/order should be correct, unless I'm misinterpreting something. 

My use-case of only one regressor is only for comparison with a full-fletched model. If this is not how fracridge is supposed to be used, then I guess the issue can be ignored :) Though maybe a different Error/Warning could be provided.

**Comment by kendrickkay** [comment]
@arokem - it would be nice if the code would run for this (degenerate) case.   

Statistically, ridge regularization for the case of one regressor is simply going to scale down the estimated weight for that regressor.  This is still meaningful in the sense that the regularization may in fact improve generalization performance (i.e. reduce overfitting).

**Comment by arokem** [comment]
Yep! That makes sense. I'll fix it up ASAP.

**Comment by oliver-contier** [comment]
Maybe to motivate my usecase a bit: I'm predicting fmri voxel time series with a parametric modulation model.
Hence the model has one general onset regressor that models the average response to all events, and then there's a bunch of parametric modulators that explain variation around that average response. As a basic sanity check, I wanted to compare the full model's generalization performance to that of a reduced model which has only the average response regressor. In order to have a "fair" comparison, I'm optimizing the regularization parameter for the reduced model as well.

I see how this is may be an edge case and potentially out of the scope of the fracridge package. So if the abovementioned behavior is intended, I understand!

I raised the issue more out of technical confusion, because I ran into the error when using `FracRidgeRegressor` (and not when using `fracridge.fracridge`). I think whether users should be discouraged or prevented from using (fractional) ridge regression with only one regressor in the first place might be a different discussion.

Anyway, thanks for both of your comments and creating/maintaining this tremendously useful tool.
